src/main/java/nextstep/ladder/infra/RandomLineGenerator.java

```diff
@@ -13,26 +14,35 @@ public class RandomLineGenerator implements LineGenerator {
     private final ThreadLocalRandom random = ThreadLocalRandom.current();
 
     @Override
-    public List<Boolean> generate(LineLength length) {
-        List<Boolean> line = new ArrayList<>();
-        Boolean prev = false;
-
+    public List<LinePoint> generate(LineLength length) {
+        List<LinePoint> line = new ArrayList<>();
+        LinePoint current = null;
         for (int i = 0; i < length.getLength(); i++) {
-            Boolean cur = random.nextBoolean();
-            cur = toNonOverlapBridge(prev, cur);
-
-            line.add(cur);
-            prev = cur;
+            current = generateLinePoint(length, i, current);
+            line.add(current);
         }
 
         return line;
     }
 
-    private Boolean toNonOverlapBridge(boolean previous, boolean current) {
-        if (current && previous) {
-            return Boolean.FALSE;
+    private LinePoint generateLinePoint(LineLength length, int current, LinePoint previous) {
+        List<MoveDirection> possibleMoveDirection = getPossibleMoveDirectionForRight();
+        if (previous != null && previous.canMoveRight()){
+            return LinePoint.of(MoveDirection.LEFT);
+        }
+        if (current == length.getLength()-1) {
+            return LinePoint.of(MoveDirection.NON);
+        }
+        if (current == 0){
```

Comment:
전반적으로 코드 포맷팅 한 번 확인해주시면 좋을 것 같습니다.
```suggestion
        if (previous != null && previous.canMoveRight()) {
            return LinePoint.of(MoveDirection.LEFT);
        }
        if (current == length.getLength() - 1) {
            return LinePoint.of(MoveDirection.NON);
        }
        if (current == 0) {
```